### ISSUE TITLE
Check Charge Times and Mode - Clip and Max Charge Current

### ISSUE THREAD

**Issue by fboundy** [stevebuk1/pv_opt#91]
**Describe the bug**
The app isn't fully checking that the charge/discharge times and inverter mode are correct on each run.

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by solarjones** [comment]
Thanks. As I've only just switched from ReadOnly to Live modes the other day, this may be a pre-existing issue to v3.5.1.
I've not looked at the code so not sure how the desired charging Amps are calculated for the relevant time slots. It didn't appear to change the Optimum Charge Current when I lowered the Charging Power. Perhaps it needs to limit based on max charging current as a new parameter? 

My current settings:
<img width="415" alt="image" src="https://github.com/fboundy/pv_opt/assets/100320468/5b6e97f5-0e0c-46e4-865a-1ebe7e8da56a">

Also spotted some random force discharge times being set (they had the same start and end times) despite Optimise Discharge being disabled.

Not your issue but I'll have another attempt at getting AppDaemon logging working, just refuses to write a log file anywhere!
